buildSrc/src/main/kotlin/TraceDebuggerTasks.kt

```diff
@@ -0,0 +1,67 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2025 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+@file:Suppress("DEPRECATION", "UNUSED_VARIABLE")
+
+import org.gradle.api.Project
+import org.gradle.api.tasks.bundling.Jar
+import org.gradle.api.file.DuplicatesStrategy
+import org.gradle.kotlin.dsl.getByType
+import org.gradle.kotlin.dsl.register
+import org.gradle.api.plugins.JavaPluginExtension
+
+// Below are tasks that are used by the trace debugger plugin.
+// When these jars are loaded the `-Dlincheck.traceDebuggerMode=true` VM argument is expected
+fun Project.registerTraceDebuggerTasks() {
+    // Ensure the Java plugin is applied (for sourceSets and runtimeClasspath)
+    plugins.apply("java")
+
+    val javaPluginExtension = extensions.getByType<JavaPluginExtension>()
+    val mainSourceSet = javaPluginExtension.sourceSets.getByName("main")
+
+    val runtimeClasspath = configurations.getByName("runtimeClasspath")
+
+    val traceDebuggerJar = tasks.register<Jar>("traceDebuggerJar") {
+        archiveBaseName.set("fat-trace-debugger")
+        duplicatesStrategy = DuplicatesStrategy.EXCLUDE
+
+        dependsOn("bootstrapJar")
+
+        // Include compiled sources
+        from(mainSourceSet.output)
+
+        // Include runtime dependencies (ASM, ByteBuddy, etc.)
+        from({
+            runtimeClasspath.resolve().filter { it.name.endsWith(".jar") }.map {
+                if (it.isDirectory) it else zipTree(it)
+            }
+        })
+
+        manifest {
+            attributes(
+                mapOf(
+                    "Premain-Class" to "org.jetbrains.kotlinx.lincheck.transformation.TraceDebuggerAgent",
+                    "Can-Redefine-Classes" to "true",
+                    "Can-Retransform-Classes" to "true"
+                )
+            )
+        }
+    }
+
+    val traceDebuggerJarNoDeps = tasks.register<Jar>("traceDebuggerJarNoDeps") {
```

Comment:
Please indicate when this task is needed and how to use the artifact

Reply:
This task was created by Ivan, he was using it for the debugging purposes. I don't know how or why 🤷

Reply:
added comment from original code